Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,309 @@
+                            break;
+                        case DeviceInputType.SecondaryButtonPress:
+                            button = controllerHandedness == MixedReality.Input.Handedness.Left ? ControllerInput.Y_Button : ControllerInput.B_Button;
+                            break;
+                        case DeviceInputType.TouchpadTouch:
+                        case DeviceInputType.TouchpadPress:
+                            button = ControllerInput.Touchpad;
+                            break;
+                        case DeviceInputType.Menu:
+                            button = ControllerInput.Menu;
+                            break;
+                        case DeviceInputType.ThumbStickTouch:
+                        case DeviceInputType.ThumbStickPress:
+                            button = ControllerInput.Thumbstick;
+                            break;
+                        default:
+                            return;
+                    }
+
+
+                    var buttonData = controllerState.CurrentReading.GetPressedValue(button);
+                    interactionMapping.BoolData = buttonData > 0.0f;
+                }
+
+                // If our value changed raise it.
+                if (interactionMapping.Changed)
+                {
+                    // Raise input system event if it's enabled
+                    if (interactionMapping.BoolData)
+                    {
+                        CoreServices.InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                    }
+                    else
+                    {
+                        CoreServices.InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                    }
+                }
+            }
+        }
+
+        private static readonly ProfilerMarker UpdateSingleAxisDataPerfMarker = new ProfilerMarker("[MRTK] HPController.UpdateSingleAxisData");
+
+        /// <summary>
+        /// Update an interaction float data type from a SingleAxis (float) input
+        /// </summary>
+        /// <remarks>
+        /// Raises a FloatInputChanged event when the float data changes
+        /// </remarks>
+        internal virtual void UpdateSingleAxisData(MixedRealityInteractionMapping interactionMapping, MotionControllerState controllerState)
+        {
+            using (UpdateSingleAxisDataPerfMarker.Auto())
+            {
+                Debug.Assert(interactionMapping.AxisType == AxisType.SingleAxis);
+                // Update the interaction data source
+                switch (interactionMapping.InputType)
+                {
+                    case DeviceInputType.TriggerPress:
+                        var triggerData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Trigger);
+                        interactionMapping.BoolData = !Mathf.Approximately(triggerData, 0.0f);
```

Comment:
This feels like it'd match to `TriggerTouch` more than `TriggerPress`

Reply:
I've always thought of `Touch` being "pressed at all" and `Press` being "fully pressed"

Reply:
Ah I see. I was slightly confused about what to do about the Trigger Touch interaction mapping since it seems like the HP controllers don't have capacitive sensors: https://www.windowscentral.com/hp-reverb-g2-controllers

Reply:
Yeah, that's true. We sorta overloaded that term for controllers without capacitive sensors to mean "pressed at all".

Reply:
The whole interaction mapping is dynamic and scalable to provide what controllers are able to offer from multiple sources.
We try to balance what the controller provides from what projects actually use.  If a project only uses a digital style interface, such as a button on the grip, then we provide that.  But well also translate that to a scale (which granted only provides 0 or 1) for projects that use a single axis float range.

It's up to each provider to determine which is right, which are altered by the needs of the project consuming the framework.

I hope that helps.

WIth WMR, we provide:
*  Digital (trigger press) - is pressed
* Digital (touch) is the trigger pressed at all
* Single Axis (press) - range of 0 and 1

As well as others, this balances all the controls and types that are exposed.  But obviously, for each project, they may only want one.